### DISCUSSION TITLE
Validation not working with shouldUnregister and conditional field

### DISCUSSION THREAD

**Discussion by AkariNishii** [react-hook-form/react-hook-form#4940] [Q&A] [answered]
When I use `shouldUnregister`, validation is not working after you trigger a check for a conditional field.

Test: https://codesandbox.io/s/validation-not-working-with-shouldunregister-cpjxy?file=/src/index.js

Step to reproduce the error:
1. Enter test longer than 3 letters in "First name" field
2. Click checkbox
3. Click "submit" button

and you can submit form, which should be prevented by validation for length of "First name" field in line 19.

Is this an expected behaviour, or am I using `shouldUnregister` in a wrong way?

**Comment by bluebill1049** [comment, marked as answer]
This is a bug.

**Reply by AkariNishii** [reply]
I see. Then this might be related - if I set `mode: 'onChange', shouldUnregister: true` for `useForm` with validation, errors not go away once it's triggered.

Test: https://codesandbox.io/s/error-does-not-go-away-with-shouldunregister-7o0pn

Step to reproduce the error:

1.  Enter longer than 3 letters as firstName input and you see the input field turns red since its validation failed
2. Remove letters until its length became less than 3
3. You still see the field is red because the error does not go away

You can check normal behaviour by setting `shouldUnregister: false` and you'll see that the error goes away on step 2.
Maybe there's some issue when removing ref by `shouldUnregister`?

**Reply by bluebill1049** [reply]
thanks, all issues related to shouldUnregister, i think I found a good solution now. patching it

**Reply by AkariNishii** [reply]
Thank you a lot!

**Reply by bluebill1049** [reply]
thank you for the reporting.
